lib/sensors/inverter_current.cpp

```diff
@@ -0,0 +1,38 @@
+#include <inverter_current.hpp>
+#include "adc_mux.hpp"
+
+namespace hyped::sensors {
+
+std::optional<InverterCurrent> InverterCurrent::create(core::ILogger &logger,
+                                                       std::shared_ptr<io::II2c> i2c,
+                                                       const std::uint8_t adc_mux_channel)
+{
+  return InverterCurrent(logger, i2c, adc_mux_channel);
+}
+
+InverterCurrent::InverterCurrent(core::ILogger &logger,
+                                 std::shared_ptr<io::II2c> i2c,
+                                 const std::uint8_t adc_mux_channel)
+    : logger_(logger),
+      i2c_(i2c),
+      adc_mux_channel_(adc_mux_channel)
+{
+}
+
+InverterCurrent::~InverterCurrent()
+{
+}
+
+std::optional<core::Float> InverterCurrent::readCurrent()
+{
+  const auto inverter_current = i2c_->readByte(kAdcMuxAddress, adc_mux_channel_);
+  if (!inverter_current) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read inverter current");
+    return std::nullopt;
+  }
+  logger_.log(core::LogLevel::kDebug, "Successfully read inverter current");
```

Comment:
this log isn't necessary, it would be pretty spammy

Reply:
I have removed it, but similar logs are in other sensors if you want me to also remove those?

Reply:
Leave them for now but we're going to need to do a big cleanup at some point soon